### PR TITLE
support night theme

### PR DESCRIPTION
**Pull request by onlymash**
### Type of changes
* [ ] Bugfix
* [x] New feature
* [ ] Translation
* [ ] General refinement
* Other:

### Details
上游的 PreferenceFixTheme 还要改回来，不然暗色主题下配置界面的 popup 背景会锁定白色，字体也是白色。

### CODE REVIEW

**Comment by madeye** [comment]
Please pass all checks first.

**Comment by Mygod** [comment]
Also I'd suggest fixing the issues in our dependencies as well. We are not
going to merge it if it still has inconsistency issues.

**Comment by onlymash** [comment]
变量命名方式是参考既有代码，不符合规范不是窝的锅。另外为什么会提示依赖版本不一致就不清楚了

File: mobile/build.gradle [hunk 1]
```diff
@@ -64,7 +64,7 @@ dependencies {
     implementation "com.google.android.gms:play-services-vision:$playServicesVersion"
     implementation "com.google.firebase:firebase-config:$playServicesVersion"
     implementation 'com.j256.ormlite:ormlite-android:5.1'
-    implementation 'com.mikepenz:materialdrawer:6.0.7'
+    implementation 'co.zsmb:materialdrawer-kt:1.3.4'
```
Comment (Mygod) [inline mobile/build.gradle]: Why is this change needed? Also I think the Java API is pretty reasonable and I don't feel like having a third-party wrapper. I prefer to reverting this change.

File: mobile/src/main/java/com/github/shadowsocks/App.kt [hunk 2]
```diff
@@ -163,6 +163,15 @@ class App : Application() {
...
         updateNotificationChannels()
+
+        val nightMode = DataStore.nightMode
+        when (nightMode) {
+            Key.nightModeSystem -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM)
+            Key.nightModeAuto -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_AUTO)
+            Key.nightModeOff -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
+            Key.nightModeOn -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
+            else -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
+        }
```
Comment (Mygod) [inline mobile/src/main/java/com/github/shadowsocks/App.kt]: I don't think @madeye likes to have a configuration for UI options. I am okay with this change.

File: plugin/src/main/res/values/styles.xml [hunk 3]
```diff
@@ -1,17 +1,28 @@
...
+        <item name="material_drawer_primary_icon">@color/drawer_primary_icon</item>
+        <item name="material_drawer_secondary_text">@color/drawer_secondary_text</item>
+        <item name="material_drawer_hint_text">@color/drawer_hint_text</item>
+        <item name="material_drawer_divider">@color/drawer_divider</item>
+        <item name="material_drawer_selected">@color/drawer_selected</item>
+        <item name="material_drawer_selected_text">@color/drawer_selected_text</item>
+        <item name="material_drawer_header_selection_text">@color/drawer_header_selection_text</item>
     </style>
-    <style name="Theme.AppCompat.Translucent" parent="Theme.AppCompat.Dialog">
+    <style name="Theme.AppCompat.Translucent" parent="Theme.AppCompat.DayNight.Dialog">
```
Comment (Mygod) [inline plugin/src/main/res/values/styles.xml]: Translucent styles don't need a color. Please revert this.

**Comment by Mygod** [comment]
You can ignore the codacy check. You could also write Java code and we will take care of the conversion.

**Comment by Mygod** [comment]
Thanks for your updates. I took another look and it seems like using `AboutFragment` screws up night mode for some reason.

I think there're plenty of reasons to believe that night mode isn't officially supported. Everything else looks reasonable good so far.

**Comment by onlymash** [comment]
你们不想支持就不不支持吧，但是别轻易砍掉上游代码 simplemenu 对的夜间模式的支持啊。simplemenu 应该是 rikka 写的，这个开发者窝了解，他的代码非常优秀，他的 app 也都支持夜间主题，没必要改啊。夜间主题默认是跟随系统的，即使没有设置，所以不想 simplemenu 受到系统影响，初始化的时候改一下默认值就可以了 AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)。

**Comment by madeye** [comment]
@Mygod Any other suggest?

@onlymash Please help to resolve the conflict.

**Comment by Mygod** [comment]
* https://github.com/Gericop/Android-Support-Preference-V7-Fix/pull/149;
* Day night settings also have to propagate to the plugins that may also use UIs;
* `AboutFragment` inconsistency issues. (or is it just me?)

**Comment by madeye** [comment]
Hmm, why we just don't support night theme? Since it needs so many changes...

**Comment by Mygod** [comment]
I think it's still manageable. I'm more worried about inconsistency issues. I can already see people creating tons of new issues regarding this feature once we merged it.

**Comment by madeye** [comment]
Yes, it's problematic. So I think we should not merge this until there is an official night mode support from Android OS>

**Comment by Mygod** [comment]
But still, very cool feature. Very fun to play around with. Adventurous people could feel free to merge this change.

**Comment by onlymash** [comment]
大概看了一下插件实现的方式，插件的配置界面好像是在插件里生成的布局，那么要想插件的设置界面也跟app保持一致，插件要重写，引入 daynight 支持，并且在启动插件配置界面的时候传递 app 的主题状态给插件。

**Comment by Mygod** [comment]
Correct. Plugins also (hopefully) use the same `plugin` library as the one in the repo. So what's needed is just an interface for passing day night settings.

**Comment by Mygod** [comment]
Oh shit I pushed the wrong stuff. Ah well. :)

@onlymash Could you push to your master branch again so that I can reopen this PR?

**Comment by Mygod** [comment]
Anyway I pushed my changes to https://github.com/Mygod/shadowsocks-android/tree/night for now.

**Comment by Mygod** [comment]
Sorry I forgot to merge this. This should have been merged after 36c72e20f55e17f903806310d78c0a28f2bf5c1e.
